### DISCUSSION TITLE
The code after "await" is not executed

### DISCUSSION THREAD

**Discussion by rozkomnadzor** [agronholm/apscheduler#805] [Q&A] [answered]
Why does my "job" complete immediately after executing one "await"?

```python
async def main(queue_name):
    async def on_message(message: DeliveredMessage):
        ...

        try:
            await bot.send_message(msg.phone, msg.message)
            # exit
            # print('some') - do not work
        except RPCError as rpc:
            ...
        finally:
            # do not work

    connection = await aiormq.connect("amqp://guest:guest@localhost/")
    async with connection:
        channel = await connection.channel()

        declare_queue = await channel.queue_declare(queue_name)

        await channel.basic_consume(
            declare_queue.queue, on_message, no_ack=True
        )
       ...
       # code is working
       await channel.basic_publish(...)
       # print('some') next code do not work again


if __name__ == '__main__':
    scheduler = AsyncIOScheduler()
    scheduler.add_job(main, "interval", args=('test_queue',), seconds=10)
    scheduler.start()
```

**Comment by agronholm** [comment, marked as answer]
There is nothing here keeping an event loop running, so I don't really understand how this would work at all.

**Reply by rozkomnadzor** [reply]
I removed most of the code that is not related to the library

**Reply by agronholm** [reply]
What am I supposed to make of the snippet then, if it's not even supposed to work? I can't exactly run it here and determine what the problem is.

**Reply by rozkomnadzor** [reply]
if you go into details, everything starts like this
```
from apscheduler.schedulers.asyncio import AsyncIOScheduler

from pyrogram import Client

app = Client("my_account")


async def job():
    await app.send_message("me", "Hi!")


scheduler = AsyncIOScheduler()
scheduler.add_job(job, "interval", seconds=3)

scheduler.start()
app.run()
```
[source](https://docs.pyrogram.org/topics/scheduling)

**Reply by agronholm** [reply]
Can you construct a minimal working example that demonstrates the issue? Something I could run too?

**Reply by rozkomnadzor** [reply]
https://pastebin.com/gytBfgBw

this code should work and is similar in meaning

**Reply by agronholm** [reply]
Except that is requires a bunch of dependencies, plus an API key. Why not just use APScheduler on its own? Or if you're suspecting that only the first "await" is executed, as the topic implies, why not add a couple `await sleep(0.1)` calls and print a message to the console or something?

**Reply by rozkomnadzor** [reply]
I found that this error is related to the "aiormq" library, for some unknown reason it stops the job. 

I checked it, as you said, with the help of sleep. I don’t receive any messages, all the code works, thanks, I’ll look for a library that will work correctly
